plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiPartitionStatsIndexSupport.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hudi.query.index;
+
+import io.airlift.log.Logger;
+import io.trino.plugin.hudi.util.TupleDomainUtils;
+import io.trino.spi.predicate.TupleDomain;
+import org.apache.hudi.avro.model.HoodieMetadataColumnStats;
+import org.apache.hudi.common.model.FileSlice;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.hash.ColumnIndexID;
+import org.apache.hudi.metadata.HoodieTableMetadata;
+import org.apache.hudi.metadata.HoodieTableMetadataUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.trino.plugin.hudi.util.TupleDomainUtils.hasSimpleNullCheck;
+
+public class HudiPartitionStatsIndexSupport
+        extends HudiColumnStatsIndexSupport
+{
+    private static final Logger log = Logger.get(HudiColumnStatsIndexSupport.class);
+
+    public HudiPartitionStatsIndexSupport(HoodieTableMetaClient metaClient)
+    {
+        super(log, metaClient);
+    }
+
+    @Override
+    public Map<String, List<FileSlice>> lookupCandidateFilesInMetadataTable(HoodieTableMetadata metadataTable, Map<String, List<FileSlice>> inputFileSlices, TupleDomain<String> regularColumnPredicates)
+    {
+        return Map.of();
```

Comment:
A bit confusing, does it mean prune all or prune nothing? How does the caller behave if we return empty map here?

Reply:
Updated this method to throw an `UnsupportedOperationException` directly - aligning with Hoodie's Spark implementation.